src/config/mod.rs

```diff
@@ -1986,6 +1986,21 @@ impl RaftEngineConfig {
         Ok(())
     }
 
+    fn optimize_for(&mut self, raft_store: &RaftstoreConfig, raft_kv_v2: bool) {
+        if !raft_kv_v2 {
+            let cur_batch_compression_thd = self.config().batch_compression_threshold;
+            // As the async-io is enabled by default (raftstore.store_io_pool_size == 1),
+            // testing records shows that using 4kb as the default value can achieve
+            // better performance and reduce the IO overhead.
+            // Meanwhile, the batch_compression_threshold cannot be modified dynamically if
+            // the threads count of async-io are changed manually.
+            self.mut_config().batch_compression_threshold = RaftEngineReadableSize(std::cmp::max(
```

Comment:
So does this means explicitly set config will also be changed? E.g, if user set the `batch-compression-threshold` to 16kb, it will be optimized to 8kb, is this expected behavior?
I still think we better make the logic straight and only do adjust if it is the default config.

Reply:
Yep.

IMO, it's expected. As more async-io threads are set, smaller the batch of raft logs will be, the `batch-compression-threshold` should be adaptive to the size of `async-io` threads.

Reply:
Why not only set it when it isn't specified explicitly in config. `4kb` may not be better for all cases.

Reply:
Then, how about introducing a flag `raft_engine.customed`, which will be true if the user manually set `raft-engine.batch-compression-ratio` with a specific value ?

And it can be combined with the current implementation that the `batch-compression-threshold` will be adaptive to the size of async-ios if the user does not set it on purpose.